### PR TITLE
Add podspec for Client and refactor dependencies

### PR DESCRIPTION
**Pull request by b-onc**
Tested in a sample project

### CODE REVIEW

File: StreamChatCore.podspec [hunk 1]
```diff
@@ -17,9 +17,8 @@ Pod::Spec.new do |spec|
 
   spec.framework = "Foundation", "UIKit"
 
+  spec.dependency "StreamChatClient"
   spec.dependency "RxSwift", "~> 5.0.0"
   spec.dependency "RxAppState", "~> 1.6.0"
```
Comment (buh) [inline StreamChatCore.podspec]: We don't use `RxAppState` anymore.

File: StreamChatClient.podspec [hunk 2]
```diff
@@ -0,0 +1,22 @@
+Pod::Spec.new do |spec|
+  spec.name = "StreamChatClient"
+  spec.version = "1.5.5"
```
Comment (buh) [inline StreamChatClient.podspec]: `2.0.0`

Comment (b-onc) [reply StreamChatClient.podspec]: Let's make it `2.0.0` when we release it (alongside StreamChat and StreamChatCore, both are 1.5.5 now)

Comment (buh) [reply StreamChatClient.podspec]: But why? It's in a separate branch and will be anyway 2.0.0. The only important thing is where the git tag `2.0.0` will be created.

Comment (b-onc) [reply StreamChatClient.podspec]: I thought it'd be better to bump all 3 together to 2.0.0 ¯\_(ツ)_/¯

File: StreamChatCore.podspec (hunk 1 above)
Comment (b-onc) [reply StreamChatCore.podspec]: Updated, we need `RxCocoa` though

File: StreamChatClient.podspec (hunk 2 above)
Comment (b-onc) [reply StreamChatClient.podspec]: I've updated it as `2.0.0`

**Pull request merged**